src/main/java/stream/flarebot/flarebot/util/general/GuildUtils.java

```diff
@@ -0,0 +1,302 @@
+            } else {
+                if (Getters.getGuildById(guildId) != null) {
+                    tmp = Getters.getGuildById(guildId).getMembers().stream()
+                            .map(Member::getUser)
+                            .filter(user -> user.getName().equalsIgnoreCase(s))
+                            .findFirst().orElse(null);
+                } else
+                    tmp = null;
+            }
+            if (tmp != null) return tmp;
+            try {
+                long l = Long.parseLong(s.replaceAll("[^0-9]", ""));
+                if (guildId == null || guildId.isEmpty()) {
+                    tmp = Getters.getUserById(l);
+                } else {
+                    Member temMember = Getters.getGuildById(guildId).getMemberById(l);
+                    if (temMember != null) {
+                        tmp = temMember.getUser();
+                    }
+                }
+                if (tmp != null) {
+                    return tmp;
+                } else if (forceGet) {
+                    return Getters.retrieveUserById(l);
+                }
+            } catch (NumberFormatException | NullPointerException ignored) {
+            }
+        }
+        return null;
+    }
+    //Channel
+
+    /**
+     * Gets a {@link TextChannel} from a string
+     *
+     * @param arg The string to get the channel from
+     * @return null if the channel couldn't be found otherwise a {@link TextChannel}
+     */
+    public static TextChannel getChannel(String arg) {
+        return getChannel(arg, null);
+    }
+
+    /**
+     * Gets a {@link TextChannel} from a string
+     *
+     * @param channelArg The string to get the channel from
+     * @param wrapper The Guild wrapper for the {@link Guild} that you want to get the channel from
+     * @return null if the channel couldn't be found otherwise a {@link TextChannel}
+     */
+    public static TextChannel getChannel(String channelArg, GuildWrapper wrapper) {
+        try {
+            long channelId = Long.parseLong(channelArg.replaceAll("[^0-9]", ""));
+            return wrapper != null ? wrapper.getGuild().getTextChannelById(channelId) : Getters.getChannelById(channelId);
+        } catch (NumberFormatException e) {
+            if (wrapper != null) {
+                List<TextChannel> tcs = wrapper.getGuild().getTextChannelsByName(channelArg, true);
+                if (!tcs.isEmpty()) {
+                    return tcs.get(0);
+                }
+            }
+            return null;
+        }
+    }
+    //Emote
+
+    /**
+     * Gets an {@link Emote}
+     *
+     * @param l the id of the emote
+     * @return null if the id is invalid otherwise a {@link Emote}
+     */
+    public static Emote getEmoteById(long l) {
+        return Getters.getGuilds().stream().map(g -> g.getEmoteById(l))
+                .filter(Objects::nonNull).findFirst().orElse(null);
+    }
+
+    /**
+     * Gets weather or not the bot can change nick
```

Comment:
Note this checks for NICKNAME_CHANGE permissions and if it doesn't have them it will check for NICKNAME_MANAGE